backend/financial_doc_processor.py

```diff
@@ -180,3 +156,222 @@ def validate_document_paths(document_paths: Dict[str, Dict[str, str]]) -> bool:
+        equity_name (str): Name of the equity (e.g., 'AAPL')
+        financial_type (str): Type of financial document (e.g., '10-K')
+    
+    Returns:
+        dict: Nested dictionary structure for upload_to_blob function
+        
+    Example:
+        >>> path = 'pdf/10-K_AAPL_summary.pdf'
+        >>> create_document_paths(path, 'AAPL', '10-K')
+        {
+            'AAPL': {
+                '10-K': 'pdf/10-K_AAPL_summary.pdf'
+            }
+        }
+    """
+    return {
+        equity_name: {
+            financial_type: output_path
+        }
+    }
+
+
+class BlobStorageManager:
+    def __init__(self):
+        self.blob_service_client = BlobServiceClient.from_connection_string(BLOB_CONNECTION_STRING)
+        self.container_client = self.blob_service_client.get_container_client(BLOB_CONTAINER_NAME)
+        self.blob_base_folder = 'financial'
+
+    def download_documents(self, equity_name: str,
+                         financial_type: str = '10-K',
+                         exclude_summary: bool = True,
+                         local_data_path: str = PDF_PATH) -> List[str]:
+        downloaded_files = []
+        try:
+            os.makedirs(local_data_path, exist_ok=True)
+            base_path = f"{self.blob_base_folder}/{financial_type}"
+            
+            # List all blobs in the financial type directory
+            all_blobs = list(self.container_client.list_blobs(name_starts_with=base_path))
+            
+            # Filter for exact equity name matches using regex pattern
+            import re
+            equity_pattern = re.compile(
+                f"{re.escape(base_path)}/{re.escape(equity_name)}(_summary)?\.pdf$"
+            )
+            
+            filtered_blobs = [
+                blob for blob in all_blobs 
+                if equity_pattern.match(blob.name) and
+                (not exclude_summary or "_summary" not in blob.name)
+            ]
+
+            logger.info(f"Found {len(filtered_blobs)} matching documents for {equity_name}")
+            
+            for blob in filtered_blobs:
+                try:
+                    logger.info(f'Downloading {blob.name}')
+                    blob_client = self.container_client.get_blob_client(blob.name)
+                    file_name = f'{financial_type}_{os.path.basename(blob.name)}'
+                    local_file_path = os.path.join(local_data_path, file_name)
+                    
+                    with open(local_file_path, "wb") as file:
+                        data = blob_client.download_blob()
+                        file.write(data.readall())
+                    
+                    downloaded_files.append(local_file_path)
+                    logger.info(f"Successfully downloaded: {file_name}")
+
+                except Exception as e:
+                    logger.error(f"Error downloading {blob.name}: {str(e)}")
+                    continue
+
+        except Exception as e:
```

Comment:
Many except Exception as e: blocks simply log the error. Consider handling known error conditions more explicitly (e.g., if a file is not found vs. a permissions issue).
Re-raising exceptions or returning None on error is okay, but consider raising custom exceptions or returning structured error information if the caller needs to distinguish different failure modes.